subject.py

```diff
@@ -0,0 +1,166 @@
+import json
+from dataclasses import dataclass, field
+from datetime import date
+from recording import RecordingInterface
+from typing import List
+
+
+@dataclass
+class Round:
+    """
+    Holds the information about the game round
+
+    This class holds information about current round and all potential rounds
+    """
+
+    allowed_stages: List = field(default_factory=lambda: ['Baseline', 'Game', 'SingleChoiceGame', 'Stimulation'])
+    current_round: int = 1 or str
+    iap: str = '1'
+    max_rounds: int = 81
+    stage: str = 'Game'
+
+    def increment(self) -> None:
+        """
+        Increment the round by 1 or reset to 1 if we hit the limit
+
+        Returns:
+            None
+        """
+        self.current_round = self.current_round + 1
+
+    def set_stage(self, stage: str) -> None:
+        """
+        Used to set the stage operation
+
+        Args:
+            stage (str): The new stage to set
+
+        Returns:
+            None
+        """
+
+        if stage not in self.allowed_stages:
+            raise ValueError(f'"{stage}" is not an accepted stage for this game')
+
+        self.stage = stage
+
+
+@dataclass
+class Subject:
+    """
+    The test subject
+
+    Holds information about the subject, the date the test occurred and holds
+    the data for the test
+    """
+
+    id: str
+    date: str
+    data: dict
+
+    def get_key(self):
+        """
+        Gets the key for the subject file name
+
+        Returns:
+            str: The subject key
+        """
+
+        return f'{self.id}({self.date})'
+
+    def create_data_template(self):
+        """
+        Create the data template ready for data population
+
+        Returns:
+            dict: The template for the json data
+        """
+
+        self.data = {
+            self.id: {
+                self.date: {
+                    'Timestamps': {
+                        'Baseline': {},
+                        'Stimulation': {}
+                    }
+                }
+            }
+        }
+
+    def add_timestamp(self, current_round: Round, words, stage, timestamp: str) -> None:
```

Comment:
With this one, we don't need to pass `stage` because we already have the `Round` which we can use to get the current stage.

We also shouldn't need to pass `words` (which isn't very descriptive), because if we are keeping `Round.current_round` and `Round.iap` up to date we can just use those instead.

Reply:
There is a weird bug that occurs if you do not pass, where it will load all 40 whitespace rounds at once and all 20 image rounds right after, ignoring the timer. Could not find reference to why that is

Reply:
It happens wayyyyy less frequently with the pass structure

Reply:
That will likely be happening outside the method, best fix Thursday onwards is just to trace what is really happening and fixing the problem at the root cause (when you're not on a deadline)